apinetlet/controllers/networkpolicy_controller.go

```diff
@@ -0,0 +1,595 @@
+				EndPort:  port.EndPort,
+			})
+		}
+
+		for _, from := range ingress.From {
+			if from.IPBlock != nil {
+				rule.CIDRBlock = append(rule.CIDRBlock, *from.IPBlock)
+			}
+
+			if from.ObjectSelector != nil {
+				ips, err := r.processObjectSelector(ctx, np, from.ObjectSelector)
+				if err != nil {
+					return nil, err
+				}
+				rule.ObjectIPs = append(rule.ObjectIPs, ips...)
+			}
+		}
+		rules = append(rules, rule)
+	}
+
+	return rules, nil
+}
+
+func (r *NetworkPolicyReconciler) parseEgressRules(ctx context.Context, np *apinetv1alpha1.NetworkPolicy) ([]apinetv1alpha1.Rule, error) {
+	var rules []apinetv1alpha1.Rule
+
+	for _, egress := range np.Spec.Egress {
+		rule := apinetv1alpha1.Rule{}
+		for _, port := range egress.Ports {
+			rule.NetworkPolicyPorts = append(rule.NetworkPolicyPorts, apinetv1alpha1.NetworkPolicyPort{
+				Protocol: port.Protocol,
+				Port:     port.Port,
+				EndPort:  port.EndPort,
+			})
+		}
+
+		for _, to := range egress.To {
+			if to.IPBlock != nil {
+				rule.CIDRBlock = append(rule.CIDRBlock, *to.IPBlock)
+			}
+
+			if to.ObjectSelector != nil {
+				ips, err := r.processObjectSelector(ctx, np, to.ObjectSelector)
+				if err != nil {
+					return nil, err
+				}
+				rule.ObjectIPs = append(rule.ObjectIPs, ips...)
+			}
+		}
+		rules = append(rules, rule)
+	}
+
+	return rules, nil
+}
+
+func (r *NetworkPolicyReconciler) processObjectSelector(ctx context.Context, np *apinetv1alpha1.NetworkPolicy, objectSelector *apinetv1alpha1.ObjectSelector) ([]apinetv1alpha1.ObjectIP, error) {
+	switch objectSelector.Kind {
+	case "NetworkInterface":
+		return r.fetchIPsFromNetworkInterfaces(ctx, np, objectSelector)
+	case "LoadBalancer":
+		return r.fetchIPsFromLoadBalancers(ctx, np, objectSelector)
+	//TODO: add more objects selector support if needed
+	default:
+		return nil, fmt.Errorf("unsupported object kind: %s", objectSelector.Kind)
+	}
+}
+
+func (r *NetworkPolicyReconciler) fetchIPsFromNetworkInterfaces(ctx context.Context, np *apinetv1alpha1.NetworkPolicy, objectSelector *apinetv1alpha1.ObjectSelector) ([]apinetv1alpha1.ObjectIP, error) {
+	sel, err := metav1.LabelSelectorAsSelector(&objectSelector.LabelSelector)
+	if err != nil {
+		return nil, err
+	}
+
+	nicList := &apinetv1alpha1.NetworkInterfaceList{}
+	if err := r.List(ctx, nicList,
+		client.InNamespace(np.Namespace),
+		client.MatchingLabelsSelector{Selector: sel},
+		client.MatchingFields{apinetclient.NetworkInterfaceSpecNetworkRefNameField: np.Spec.NetworkRef.Name},
+	); err != nil {
+		return nil, fmt.Errorf("error listing apinet network interfaces: %w", err)
+	}
+
+	var ips []apinetv1alpha1.ObjectIP
+
+	for _, nic := range nicList.Items {
+		if nic.Status.State != apinetv1alpha1.NetworkInterfaceStateReady {
+			continue
+		}
+
+		for _, ip := range nic.Spec.IPs {
+			ips = append(ips, apinetv1alpha1.ObjectIP{
+				Prefix:   net.IPPrefix{Prefix: netip.PrefixFrom(ip.Addr, 32)},
+				IPFamily: corev1.IPv4Protocol, // TODO: later support for IPv6
```

Comment:
Can't we use the same `IPFamily` as the `NetworkInterface` instead of hardcoding it to IPv4?

Reply:
Done